src/ReadOnlyArrays.jl

```diff
@@ -62,4 +68,15 @@ Base.unsafe_convert(p::Type{Ptr{T}}, roa::ReadOnlyArray) where {T} =
 Base.stride(roa::ReadOnlyArray, i::Int) = stride(roa.parent, i)
 Base.parent(roa::ReadOnlyArray) = roa.parent
 
+Base.convert(::Type{ReadOnlyArray{T, N}}, mutable_arr::AbstractArray{T, N}) where {T, N} = ReadOnlyArray(mutable_arr)
+
+# Define aliases:
+const ReadOnlyVector{T, P} = ReadOnlyArray{T,1,P}
+const ReadOnlyMatrix{T,P} = ReadOnlyArray{T,2,P}
+
+# Allow construction through the alias:
+ReadOnlyVector(parent::AbstractVector) = ReadOnlyArray(parent)
```

Comment:
do you want to also allow passing parameters (as above in th constructor?)

Reply:
or in general - I am curious why e.g. `ReadOnlyVector{T}` definition is not needed, but `ReadOnlyVector` is needed.

Reply:
I'm not sure why the other overloads aren't needed, but the unit tests prove that they are not. Overload resolution is a tricky beast :P